AppiumForMac/Server/Handlers/AfMHandlers.m

```diff
@@ -543,9 +543,9 @@ - (AppiumMacHTTPJSONResponse *)post_elements:(NSString*)path data:(NSData*)postD
                     [elements addObject:[NSDictionary dictionaryWithObject:myKey forKey:@"ELEMENT"]];
                 }
                 
-                return [AppiumMacHTTPJSONResponse responseWithJson:elements status:kAfMStatusCodeSuccess session:session.sessionId];
+                return [AppiumMacHTTPJSONResponse responseWithJson:elements status:*statusCode session:session.sessionId];
             }
-            return [AppiumMacHTTPJSONResponse responseWithJson:nil status:kAfMStatusCodeNoSuchElement session:session.sessionId];
+            return [AppiumMacHTTPJSONResponse responseWithJson:matches status:*statusCode session:session.sessionId];
```

Comment:
here we can simply return an empty array. I think it makes sense to change this block to

```NSMutableArray *matches = [NSMutableArray new];
            [locator findAllUsingBaseUIElement:nil results:matches statusCode:statusCode];
            if (0 == matches.count) {
                return [AppiumMacHTTPJSONResponse responseWithJson:matches status:*statusCode session:session.sessionId];
            }

                 NSMutableArray *elements = [NSMutableArray new];
                for(PFUIElement *element in matches)
                {
                    session.elementIndex++;
                    NSString *myKey = [NSString stringWithFormat:@"%d", session.elementIndex];
                    [session.elements setValue:element forKey:myKey];
	                    [elements addObject:[NSDictionary dictionaryWithObject:myKey forKey:@"ELEMENT"]];
	         }
	         return [AppiumMacHTTPJSONResponse responseWithJson:elements status:*statusCode session:session.sessionId];
```